app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,60 @@
+class PicturesController < ApplicationController
+
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def index
+    @pictures = Picture.all
+  end
+
+  def new
+    @picture = Picture.new
+  end
+
+  def create
+    @picture = current_user.pictures.build(picture_params)
+    if @picture.save
+      redirect_to pictures_path
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @picture = Picture.find(params[:id])
+  end
+
+  def edit
+    @picture = Picture.find(params[:id])
+    if @picture.user != current_user
+      flash[:notice] = 'You did not add that picture'
+      redirect_to '/pictures'
+    end
+  end
+
+  def update
+    @picture = Picture.find(params[:id])
+    @picture.update(picture_params)
+
+    if @picture.errors.any?
+      render 'edit'
+    else
+      redirect_to '/pictures'
+    end
+  end
+
+  def destroy
+    @picture = Picture.find(params[:id])
+    if @picture.user == current_user
+      @picture.destroy
+      flash[:notice] = 'Picture deleted successfully'
+      redirect_to '/pictures'
```

Comment:
When the redirect is the same after if as it is after else, just have it after end.